data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/tekuv1/admin/Readiness.java

```diff
@@ -60,10 +79,27 @@ public Readiness(final DataProvider provider) {
   @Override
   public void handle(final Context ctx) throws Exception {
     ctx.header(Header.CACHE_CONTROL, CACHE_NONE);
-    if (!chainDataProvider.isStoreAvailable() || syncProvider.isSyncing()) {
+    if (!chainDataProvider.isStoreAvailable()
+        || syncProvider.isSyncing()
+        || !reachedTargetPeerCount(ctx)) {
       ctx.status(SC_SERVICE_UNAVAILABLE);
     } else {
       ctx.status(SC_OK);
     }
   }
+
+  private boolean reachedTargetPeerCount(final Context ctx) {
+    try {
+      final Map<String, List<String>> parameters = ctx.queryParamMap();
+      if (parameters.isEmpty()) return true; // Backwards compatibility
```

Comment:
nit: We always use braces on if statements.
```suggestion
      if (parameters.isEmpty()) {
        return true; // Backwards compatibility
      }
```